### ISSUE TITLE
ClassifierDLApproach is not Learning across Epochs

### ISSUE THREAD

**Issue by DarDevDuck** [JohnSnowLabs/spark-nlp#2442]
## ClassifierDLApproach unable to improve Accuracy with each round

Did a simple experiment where I am trying to train `sentence` ->`BertSentenceEmbeddings` ->  `label: emotion` following this data source: https://www.kaggle.com/praveengovi/emotions-dataset-for-nlp 

### Code: 
```scala
package mlmodel

import com.johnsnowlabs.nlp.annotators.classifier.dl.ClassifierDLApproach
import datamodel.{Sentence, TrainingDataSchema}
import mlmodel.PipelineStages._
import org.apache.spark.ml.{Pipeline, PipelineModel}
import org.apache.spark.sql.SparkSession

object BertSentenceEmbedding {
  def main(args: Array[String]): Unit = {
    val spark: SparkSession = SparkSession.builder.appName("BertSentenceModel").master("local[3]")
      .getOrCreate()

    import spark.implicits._
    val trainingDataSet = spark.read.option("delimiter", ";")
      .schema(TrainingDataSchema.schema)
      .csv("./src/main/resources/train.csv")
 
    val valDataSet = spark.read.option("delimiter", ";")
      .schema(TrainingDataSchema.schema)
      .csv("./src/main/resources/val.csv")

    val combineDataSets = trainingDataSet.union(valDataSet)

    val classifierDeepLearning: ClassifierDLApproach = new ClassifierDLApproach()
      .setInputCols("sentenceEmbeddings")
      .setLabelColumn("emotion")
      .setOutputCol("class")
      .setBatchSize(180)
      .setValidationSplit(0.125f)
      .setMaxEpochs(10)
      .setEnableOutputLogs(true)

    val pipeline: Pipeline = new Pipeline()
      .setStages(Array(documentAssembler("sentence", "document")
        , bertSentenceEmbeddings("sent_small_bert_L12_768", List("document"), "sentenceEmbeddings")
        , classifierDeepLearning))

    val supervisedModel: PipelineModel = pipeline.fit(combineDataSets)

    supervisedModel.write.save(s"v1_supervised_bert_sentence${System.currentTimeMillis()}")
  }
}

```

```scala
  def documentAssembler(inputColumn: String, outputColumn: String): PipelineStage = {
    new DocumentAssembler()
      .setInputCol(inputColumn)
      .setOutputCol(outputColumn)
  }

  def bertSentenceEmbeddings(model:String, inputColumns: List[String], outputColumn: String): BertSentenceEmbeddings = {
    BertSentenceEmbeddings.pretrained(model)
      .setInputCols(inputColumns.toArray)
      .setOutputCol(outputColumn)
      .setCaseSensitive(false)
  }
```

### Logs:
```
Training started - epochs: 10 - learning_rate: 0.005 - batch_size: 160 - training_examples: 15750 - classes: 6
Epoch 1/10 - 4.80s - loss: 172.64043 - acc: 0.2900145 - val_acc: 30.444443 - batches: 99
Epoch 2/10 - 3.63s - loss: 172.31572 - acc: 0.2910987 - val_acc: 30.444443 - batches: 99
Epoch 3/10 - 3.29s - loss: 172.31572 - acc: 0.2910987 - val_acc: 30.444443 - batches: 99
Epoch 4/10 - 3.17s - loss: 172.31572 - acc: 0.2910987 - val_acc: 30.444443 - batches: 99
Epoch 5/10 - 3.18s - loss: 172.31572 - acc: 0.2910987 - val_acc: 30.444443 - batches: 99
Epoch 6/10 - 3.26s - loss: 172.31572 - acc: 0.2910987 - val_acc: 30.444443 - batches: 99
Epoch 7/10 - 3.07s - loss: 172.31572 - acc: 0.2910987 - val_acc: 30.444443 - batches: 99
Epoch 8/10 - 3.12s - loss: 172.31572 - acc: 0.2910987 - val_acc: 30.444443 - batches: 99
Epoch 9/10 - 3.16s - loss: 172.31572 - acc: 0.2910987 - val_acc: 30.444443 - batches: 99
Epoch 10/10 - 3.22s - loss: 172.31572 - acc: 0.2910987 - val_acc: 30.444443 - batches: 99

```

## Expected Behavior
I thought it would be a simple "HelloWorld" exercise which can achieve 90% accuracy etc as denoted in Kaggle by some people. 
I expect the loss to be in the `0.xxx` range. I expect the accuracy to improve across epochs. Am I misunderstanding anything?
Would appreciate the help

## Your Environment
build.sbt
```
scalaVersion := "2.11.12"

libraryDependencies ++= Seq(
  "com.johnsnowlabs.nlp" %% "spark-nlp" % "2.7.3",
  "org.apache.spark" %% "spark-core" % "2.4.7",
  "org.apache.spark" %% "spark-sql" % "2.4.7",
  "org.apache.spark" %% "spark-mllib" % "2.4.7"
)
```

**Comment by maziyarpanahi** [comment]
When learning stops is an indication of a few things:

- There is not enough training examples (you can augment the dataset)
- The embeddings (transfer learning) used is not sufficient enough (choose a better/larger/etc., models)
- Your batchSize is really large
- Your learning rate is very high

I would start with making the batchSize to `8` and setLr to `1e-4f` to see what happens. After that, you can try a `UniversalSentenceEncoder` instead of `BertSentenceEmbeddings` and see the differences.

Hypretuning is important and the default values cannot be applied to all use cases.

**Comment by DarDevDuck** [comment]
Hey, Thank you. I managed to improve it slightly which kind of proves that the ClassifierDLApproach is working as intended and that I would need to tune it. Thank you.

```
Training started - epochs: 10 - learning_rate: 1.0E-4 - batch_size: 8 - training_examples: 15750 - classes: 6
Epoch 1/10 - 16.06s - loss: 3175.6267 - acc: 0.4644521 - val_acc: 53.288887 - batches: 1969
Epoch 2/10 - 14.99s - loss: 3043.4617 - acc: 0.5197112 - val_acc: 53.82222 - batches: 1969
Epoch 3/10 - 14.87s - loss: 3019.4517 - acc: 0.5244326 - val_acc: 53.955555 - batches: 1969
Epoch 4/10 - 14.88s - loss: 3008.8796 - acc: 0.52614754 - val_acc: 54.08889 - batches: 1969
Epoch 5/10 - 15.13s - loss: 3002.7412 - acc: 0.5276084 - val_acc: 54.17778 - batches: 1969
Epoch 6/10 - 15.16s - loss: 2998.6455 - acc: 0.5290693 - val_acc: 54.4 - batches: 1969
Epoch 7/10 - 14.72s - loss: 2995.6704 - acc: 0.52983147 - val_acc: 54.355556 - batches: 1969
Epoch 8/10 - 14.87s - loss: 2993.3562 - acc: 0.530022 - val_acc: 54.311108 - batches: 1969
Epoch 9/10 - 14.70s - loss: 2991.4905 - acc: 0.5308477 - val_acc: 54.22222 - batches: 1969
Epoch 10/10 - 14.86s - loss: 2989.9512 - acc: 0.53122884 - val_acc: 54.311108 - batches: 1969
```
